modules/apps/portal-remote/portal-remote-cors-impl/src/test/java/com/liferay/portal/remote/cors/internal/url/pattern/mapper/StaticSizeTrieURLPatternMapperTest.java

```diff
@@ -14,14 +14,30 @@
 
 package com.liferay.portal.remote.cors.internal.url.pattern.mapper;
 
+import com.liferay.portal.kernel.util.HashMapBuilder;
+
 import java.util.Map;
 
+import org.junit.Assert;
+import org.junit.Test;
+
 /**
  * @author Brian Wing Shun Chan
  */
 public class StaticSizeTrieURLPatternMapperTest
 	extends SimpleURLPatternMapperTest {
 
+	@Test
+	public void testGetExtensionValueBounds() {
+		URLPatternMapper<String> urlPatternMapper = createURLPatternMapper(
+			HashMapBuilder.put(
+				"*.jsp", "*.jsp"
+			).build());
+
+		Assert.assertNull(urlPatternMapper.getValue("jsp"));
```

Comment:
We probably need some sort systematic(organized) and comprehensive way to test by the specification requirements.

1. The container will try to find an exact match of the path of the request to the path of the servlet. A successful match selects the servlet.

2. The container will recursively try to match the longest path-prefix. This is done by stepping down the path tree a directory at a time, using the ’ / ’ character as a path separator. The longest match determines the servlet selected.

3. If the last segment in the URL path contains an extension (e.g. .jsp ), the servlet container will try to match a servlet that handles requests for the extension. An extension is defined as the part of the last segment after the last ’ . ’ character.

I can see this test is for part of the 3rd requirement.



Also why not more this test method to the simple? Should'nt it be a global test?

Reply:
I was not sure if this made sense for all the impementations. I am not against moving it there.

Reply:
Yeah, I think it makes for all implementation, because it's a test for specification.